rotkehlchen/usage_analytics.py

```diff
@@ -22,134 +24,114 @@
-    except JSONDecodeError:
-        return None
-
-    return GeolocationData(
-        country_code=json_ret.get('country', 'unknown'),
-        city=json_ret.get('city', 'unknown'),
-    )
-
-
-def query_ipstack() -> Optional[GeolocationData]:
-    """
-    10,000 requests per month tied to the API Key
-    https://ipstack.com/
-    """
-    try:
-        response = requests.get(
-            'http://api.ipstack.com/check?access_key=affd920d6e1008a614900dbc31d52fa6',
-            timeout=LOCATION_DATA_QUERY_TIMEOUT,
-        )
-    except requests.exceptions.RequestException:
-        return None
-
-    if response.status_code != HTTPStatus.OK:
+        user_ip = requests.get(
+            url='https://checkip.amazonaws.com',
+            timeout=DEFAULT_TIMEOUT_TUPLE,
+        ).text.strip()
+    except requests.exceptions.RequestException as e:
+        log.debug(f'Failed to get ip. {str(e)}')
         return None
 
     try:
-        json_ret = jsonloads_dict(response.text)
-    except JSONDecodeError:
-        return None
-
-    return GeolocationData(
-        country_code=json_ret.get('country_code', 'unknown'),
-        city=json_ret.get('city', 'unknown'),
-    )
-
-
-def retrieve_location_data() -> Optional[GeolocationData]:
-    # First try asking the no api key service
-    location_data = query_ipwhoisio()
-    if location_data:
-        return location_data
-
-    # Then the one with the most API requests per key
-    location_data = query_ipinfo()
-    if location_data:
-        return location_data
-
-    # finally the 3rd one as backup
-    location_data = query_ipstack()
-    # Can also be None if last query also fails
-    # TODO: Add more fallbacks
-    return location_data
+        with maxminddb.open_database(filename) as reader:
+            data = reader.get(user_ip)
+            location = data['country']['iso_code']
+            if location == 'None':
+                return None
+            return GeolocationData(country_code=location)
+    except maxminddb.errors.InvalidDatabaseError as e:
+        os.remove(filename)
+        log.debug(f'Failed to read database {str(e)}')
+    except ValueError as e:
+        log.debug(f'Wrong ip search {str(e)}')
+    return None
 
 
-def create_usage_analytics() -> Dict[str, Any]:
+def create_usage_analytics(data_dir: Path) -> Dict[str, Any]:
     analytics = {}
 
     analytics['system_os'] = platform.system()
     analytics['system_release'] = platform.release()
     analytics['system_version'] = platform.version()
     analytics['rotki_version'] = get_system_spec()['rotkehlchen']
 
-    location_data = retrieve_location_data()
+    location_data = retrieve_location_data(data_dir)
     if location_data is None:
-        location_data = GeolocationData('unknown', 'unknown')
+        location_data = GeolocationData('unknown')
 
     analytics['country'] = location_data.country_code
-    analytics['city'] = location_data.city
+    analytics['city'] = 'unknown'
```

Comment:
write in a comment we don't need this and we deprecate it. Only keeping it here for api compatibility